recipes/expat/all/conanfile.py

```diff
@@ -25,87 +27,72 @@ class ExpatConan(ConanFile):
         "char_type": "char",
     }
 
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+            copy(self, patch["patch_file"],
+                 os.path.join(self.folders.recipe_folder, "patches"),
+                 os.path.join(self.folders.expor_sources_folder, "patches"))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if tools.Version(self.version) < "2.2.8":
+        if Version(self.version) < "2.2.8":
             del self.options.char_type
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
+        if not is_msvc(self):
+            del self.settings.compiler.libcxx
```

Comment:
deleting a non existent compiler setting (tested with `msvc`) raises in Conan v2